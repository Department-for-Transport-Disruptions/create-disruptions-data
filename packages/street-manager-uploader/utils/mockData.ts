const sqsMessage = {
    permitReferenceNumber: "TSR1591199404915-01",
    highwayAuthority: "CITY OF WESTMINSTER",
    highwayAuthoritySwaCode: 5990,
    worksLocationCoordinates: "LINESTRING(501251.53 222574.64,501305.92 222506.65)",
    streetName: "HIGH STREET NORTH",
    areaName: "LONDON",
    workCategory: "Standard",
    trafficManagementType: "Road closure",
    proposedStartDateTime: "2020-06-10T00:00:00.000Z",
    proposedEndDateTime: "2020-06-12T00:00:00.000Z",
    actualStartDateTime: null,
    actualEndDateTime: null,
    workStatus: "Works planned",
    usrn: "8401426",
    activityType: "Remedial works",
    worksLocationType: "Cycleway, Footpath",
    isTrafficSensitive: "Yes",
    permitStatus: "permit_modification_request",
    town: "LONDON",
    currentTrafficManagementType: "Multi-way signals",
    currentTrafficManagementTypeUpdateDate: null,
    lastUpdatedDateTime: "2020-06-04T08:00:00.000Z",
};

export const mockSqsEvent = {
    Records: [
        {
            messageId: "id",
            receiptHandle: "test",
            body: JSON.stringify(sqsMessage),
            md5OfBody: "test",
            eventSource: "test",
            eventSourceARN: "test",
            awsRegion: "test",
            attributes: {
                ApproximateReceiveCount: "test",
                SentTimestamp: "test",
                SenderId: "test",
                ApproximateFirstReceiveTimestamp: "test",
            },
            messageAttributes: {
                message: { dataType: "String" },
            },
        },
    ],
};
