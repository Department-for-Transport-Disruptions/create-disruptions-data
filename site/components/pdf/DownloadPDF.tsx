import { Document, Page, StyleSheet } from "@react-pdf/renderer";
import { ExportDisruptions } from "../../schemas/disruption.schema";
import PDFHeader from "./PDFHeader";
import PDFRows from "./PDFRows";

export const PDFStyles = StyleSheet.create({
    page: {
        flexDirection: "column",
        fontFamily: "Helvetica",
    },
});

export interface PDFProps {
    disruptions: ExportDisruptions;
}

const PDFDoc = ({ disruptions }: PDFProps) => (
    <Document>
        <Page size="A1" style={PDFStyles.page} wrap orientation="landscape">
            <PDFHeader />
            <PDFRows disruptions={disruptions} />
        </Page>
    </Document>
);

export default PDFDoc;
