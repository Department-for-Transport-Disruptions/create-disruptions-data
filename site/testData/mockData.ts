/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { mockRequest, mockResponse } from "mock-req-res";
import { NextPageContext } from "next";
import React from "react";
import { Mock, vi } from "vitest";
import { COOKIE_ID_TOKEN, COOKIES_POLICY_COOKIE } from "../constants";

export interface GetMockContextInput {
    session?: { [key: string]: any };
    cookies?: any;
    body?: any;
    url?: any;
    uuid?: any;
    mockWriteHeadFn?: Mock<any, any>;
    mockEndFn?: Mock<any, any>;
    isLoggedin?: boolean;
    query?: any;
}

export const getMockRequestAndResponse = ({
    cookieValues = {},
    body = null,
    mockWriteHeadFn = vi.fn(),
    mockEndFn = vi.fn(),
    requestHeaders = {},
    isLoggedin = true,
    url = null,
    query = null,
}: GetMockRequestAndResponse = {}): { req: any; res: any } => {
    const res = mockResponse({ writeHead: mockWriteHeadFn, end: mockEndFn });

    const {
        idToken = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJjdXN0b206bm9jIjoiVEVTVCIsImVtYWlsIjoidGVzdEBleGFtcGxlLmNvbSIsImp0aSI6Ijg1MmQ1MTVlLTU5YWUtNDllZi1iMTA5LTI4YTRhNzk3YWFkNSIsImlhdCI6MTU5Mjk4NzMwNywiZXhwIjoxNTkyOTkwOTA3fQ.DFdxnpdhykDONOMeZMNeMUFpCHZ-hQ3UXczq_Qh0IAI",
        cookiePolicy = null,
    } = cookieValues;

    const defaultSession = {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        destroy: (): void => {},
    };

    let cookieString = "";

    cookieString += isLoggedin ? `${COOKIE_ID_TOKEN}=${idToken as string};` : "";

    cookieString += cookiePolicy ? `${COOKIES_POLICY_COOKIE}=${encodeURI(JSON.stringify(cookiePolicy))}` : "";

    const req = mockRequest({
        connection: {
            encrypted: true,
        },
        url,
        headers: {
            host: "localhost:3000",
            cookie: cookieString,
            origin: "localhost:3000",
            ...requestHeaders,
        },
        cookies: cookieValues,
        session: { ...defaultSession },
    });

    if (body) {
        req.body = body;
    }
    if (query) {
        req.query = query;
    }

    return { req, res };
};

export const getMockContext = ({
    cookies = {},
    body = null,
    uuid = {},
    mockWriteHeadFn = vi.fn(),
    mockEndFn = vi.fn(),
    isLoggedin = true,
    url = null,
    query = "",
}: GetMockContextInput = {}): NextPageContext => {
    const { req, res } = getMockRequestAndResponse({
        cookieValues: cookies,
        body,
        uuid,
        mockWriteHeadFn,
        mockEndFn,
        requestHeaders: {},
        isLoggedin,
        url,
    });

    const ctx: NextPageContext = {
        res,
        req,
        pathname: "",
        query,
        // eslint-disable-next-line react/display-name
        AppTree: () => React.createElement("div"),
    };

    return ctx;
};

export interface GetMockRequestAndResponse {
    cookieValues?: any;
    body?: any;
    uuid?: any;
    mockWriteHeadFn?: Mock<any, any>;
    mockEndFn?: Mock<any, any>;
    requestHeaders?: any;
    isLoggedin?: boolean;
    url?: any;
    query?: any;
}
