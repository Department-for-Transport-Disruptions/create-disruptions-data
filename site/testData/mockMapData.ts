export const mockMarkerData = [
    {
        atcoCode: "1800SB12581",
        commonName: "Cheapside",
        indicator: "Stop CL",
        longitude: -2.244551,
        latitude: 53.48096,
        bearing: "W",
        stopType: "BCT",
        busStopType: "MKD",
    },
    {
        atcoCode: "1800SB05801",
        commonName: "Cheapside",
        indicator: "Stop CO",
        longitude: -2.243361,
        latitude: 53.48097,
        bearing: "E",
        stopType: "BCT",
        busStopType: "MKD",
    },
    {
        atcoCode: "1800SB05751",
        commonName: "Manchester Town Hall",
        indicator: "Stop SC",
        longitude: -2.244697,
        latitude: 53.480024,
        bearing: "SE",
        stopType: "BCT",
        busStopType: "MKD",
    },
];
