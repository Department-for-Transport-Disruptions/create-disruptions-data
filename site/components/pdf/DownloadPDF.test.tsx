import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { exportDisruption } from '../../testData/mockData';
import DownloadPDF from './DownloadPDF';

describe('DownloadPDF', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<DownloadPDF disruptions={exportDisruption} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
