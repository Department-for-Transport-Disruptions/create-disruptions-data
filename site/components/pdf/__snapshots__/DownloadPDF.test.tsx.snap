// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DownloadPDF > should render correctly 1`] = `
<DOCUMENT>
  <PAGE
    size="A1"
    style={
      {
        "flexDirection": "column",
        "fontFamily": "Helvetica",
      }
    }
    wrap={true}
  >
    <VIEW
      fixed={true}
      style={
        {
          "alignItems": "flex-start",
          "borderBottomWidth": 4,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "fontFamily": "Helvetica-Bold",
          "fontStyle": "bold",
          "margin": 5,
          "textAlign": "left",
        }
      }
    >
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        id
      </TEXT>
      <TEXT
        style={
          {
            "width": "30.5%",
          }
        }
      >
        title
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        mode
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        operator wide
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        network wide
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        services affected
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        stops affected
      </TEXT>
      <TEXT
        style={
          {
            "width": "4.5%",
          }
        }
      >
        start
      </TEXT>
      <TEXT
        style={
          {
            "width": "4.5%",
          }
        }
      >
        end
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        severity
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        live
      </TEXT>
      <TEXT
        style={
          {
            "width": "8.5%",
          }
        }
      >
        status
      </TEXT>
    </VIEW>
    <VIEW
      style={
        {
          "alignItems": "flex-start",
          "borderBottomWidth": 1,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "fontFamily": "Helvetica",
          "margin": 5,
          "paddingBottom": 5,
          "textAlign": "left",
        }
      }
    >
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        0
      </TEXT>
      <TEXT
        style={
          {
            "width": "30.5%",
          }
        }
      >
        Test summary
      </TEXT>
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "4.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "4.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "6.5%",
          }
        }
      >
        Unknown
      </TEXT>
      <TEXT
        hyphenationCallback={[Function]}
        style={
          {
            "width": "6.5%",
          }
        }
      />
      <TEXT
        style={
          {
            "width": "8.5%",
          }
        }
      >
        Pending Approval
      </TEXT>
    </VIEW>
  </PAGE>
</DOCUMENT>
`;
